lldb/source/Plugins/ExpressionParser/Clang/ClangUserExpression.cpp

```diff
@@ -219,9 +219,9 @@ void ClangUserExpression::ScanContext(ExecutionContext &exe_ctx, Status &err) {
     // whatever runtime the debug info says the object pointer belongs to.  Do
     // that here.
 
-    ClangASTMetadata *metadata =
-        TypeSystemClang::DeclContextGetMetaData(decl_context, function_decl);
-    if (metadata && metadata->HasObjectPtr()) {
+    if (auto metadata = TypeSystemClang::DeclContextGetMetaData(decl_context,
```

Comment:
Nit: I know everyone has a different opinion on `auto`, but to me it's not obvious from the method name what the return type is, let alone that it's wrapped in an optional.

Reply:
Yea that's fair, i don't mind adding back the typenames